### PR TITLE
chore: cleanup tpch execute

### PR DESCRIPTION
**Pull request by EdAbati**
## What type of PR is this? (check all applicable)

- [x] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

## If you have comments or can explain your changes, please do so below.

Inspired by #1348, I thought it would be handy if adding/removing backends from TPCH was a bit easier. With this change, adding/removing a key in the `BACKEND_READ_FUNC_MAP` dictionary would add/remove a backend from the test.

While refactoring this I found some duplicated (a path was present twice) and unused code (the read functions that I commented out). I hope that with this it would be easier to spot

The diff is big because I removed all the `execute` folder. I know that this is lower priority so don't feel like you have to review straight away, it can wait :)

### CODE REVIEW

File: tpch/execute.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+REGION_PATH = DATA_DIR / "region.parquet"
+NATION_PATH = DATA_DIR / "nation.parquet"
+SUPPLIER_PATH = DATA_DIR / "supplier.parquet"
+PART_PATH = DATA_DIR / "part.parquet"
+PARTSUPP_PATH = DATA_DIR / "partsupp.parquet"
+ORDERS_PATH = DATA_DIR / "orders.parquet"
+CUSTOMER_PATH = DATA_DIR / "customer.parquet"
+
+BACKEND_READ_FUNC_MAP = {
+    # "pandas": lambda x: pd.read_parquet(x, engine="pyarrow"), # noqa: ERA001
```
Comment (EdAbati) [inline tpch/execute.py:25]: should we just remove these? I kept them because they were still in the dictionary

File: tpch/tests/queries_test.py [hunk 2]
```diff
@@ -4,20 +4,22 @@
...
-            [sys.executable, "-m", f"execute.{script_path.stem}"],
-            capture_output=True,
-            text=True,
-            check=False,
-        )
-        assert (
-            result.returncode == 0
-        ), f"Script {script_path} failed with error: {result.stderr}"
+
+@pytest.mark.parametrize("query_path", QUERIES_DIR.glob("q[1-9]*.py"))
```
Comment (EdAbati) [inline tpch/tests/queries_test.py:14]: if this PR doesn't get merged, I think that at least we should have this parametrised :)

File: tpch/execute.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    "pandas[pyarrow]": lambda x: pd.read_parquet(
+        x, engine="pyarrow", dtype_backend="pyarrow"
+    ),
+    # "polars[eager]": lambda x: pl.read_parquet(x),
+    "polars[lazy]": lambda x: pl.scan_parquet(x),
+    "pyarrow": lambda x: pq.read_table(x),
+    # "dask": lambda x: dd.read_parquet(x, engine="pyarrow", dtype_backend="pyarrow"), # noqa: ERA001
```
Comment (FBruzzesi) [inline tpch/execute.py:32]: Should we enable dask again? They fixed the issue for which we removed it in the first place.

File: tpch/tests/queries_test.py (hunk 2 above)
Comment (FBruzzesi) [reply tpch/tests/queries_test.py:14]: +1

**Review by FBruzzesi — COMMENTED** [review]
I love this kind of refactors and simplifications! Thanks @EdAbati !

I left just one comment regarding enabling running dask since they fixed the issue, and from our side we adjusted given their suggestions

File: tpch/execute.py (hunk 3 above)
Comment (EdAbati) [reply tpch/execute.py]: you are right. tested locally, it seems to be fine 👀

**Review by FBruzzesi — APPROVED** [review]
It's a +10 from me 

DogsClapGIF

**Review by MarcoGorelli — APPROVED** [review]
awesome, much better, thanks both!

**Pull request merged**